### PR TITLE
Remove automatic cleaning

### LINKED ISSUE #42: Remove clean as configuration option
Setting `clean=True` in the configuration is dangerous, because it will delete things without user input and can lead to data loss. 

~It might be a good idea to add a prompt to the `configure` command where if a user sets `clean=True`, they need to read a warning and confirm that they want to configure things that way.~ 

For now, we should just remove clean from the configuration and from the post-save hook. This isn't mature enough to let users run automatically.

### PR DESCRIPTION
**Pull request by jayqi**
- Remove `clean` command from config and automatic cleaning from the post-save hook (Closes #42)
- Put clear messaging that the `clean` command is experimental, and document limitations

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -156,7 +154,7 @@ Options:
   --help                Show this message and exit.
 
 Commands:
-  clean      Remove subfolders/files not matching .nbautoconvert...
+  clean      (EXPERIMENTAL) Remove subfolders/files not matching...
```
Comment (ejm714) [inline README.md:157]: What's the `...` for here?

Comment (jayqi) [reply README.md:157]: That's just what `typer`/`click` do when the docstring is too long and they truncate it.

Comment (ejm714) [reply README.md:157]: Ah right forgot this section is copied and pasted in

**Comment by jayqi** [comment]
Opened #46

**Comment by pjbull** [comment]
🔥 🔥

**Pull request merged**